src/components/ElementResizeListener.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {
+  useCallback, useRef, useEffect, RefObject,
+} from 'react';
+
+interface Props {
+  onResize: (event: Event) => void;
+}
+
+const ElementResizeListener: React.FC<Props> = ({ onResize }) => {
+  const rafRef = useRef(0);
+  const objectRef: RefObject<HTMLObjectElement> = useRef(null);
+  const onResizeRef = useRef(onResize);
+
+  onResizeRef.current = onResize;
+
+  const localOnResize = useCallback((e: Event) => {
+    if (rafRef.current) {
+      cancelAnimationFrame(rafRef.current);
+    }
+    rafRef.current = requestAnimationFrame(() => {
+      onResizeRef.current(e);
+    });
+  }, []);
+
+  const onLoad = useCallback(() => {
+    const obj = objectRef.current;
+    if (obj && obj.contentDocument && obj.contentDocument.defaultView) {
+      obj.contentDocument.defaultView.addEventListener('resize', localOnResize);
+    }
+  }, [localOnResize]);
+
```

Comment:
Can we somehow wrap addEventListener and removeEventListener to the same useEffect (as is commonly done in useEffects)? Like this:
```
const [isLocalResize, setLocalResize] = useState(false);

useEffect(() => {
  setLocalResize(!!obj.contentDocument.defaultView);
}, [localOnResize);

```

```
useEffect(() => {
  obj?.contentDocument?.defaultView?.removeEventListener('resize', localOnResize);
  obj?.contentDocument?.defaultView?.addEventListener.('resize', localOnResize);
  return () => {
    obj?.contentDocument?.defaultView?.removeEventListener('resize', localOnResize);
  };
}, [isLocalResize]);
```